polytiles/polytiles.py

```diff
@@ -152,49 +148,48 @@ def close(self):
-        self.con = sqlite3.connect(self.filename)
+        if not self.filename.endswith(".mbtiles"):
+            self.filename = self.filename + ".mbtiles"
+        self.con = sqlite3.connect(self.filename, isolation_level=None)
         self.cur = self.con.cursor()
-        self.cur.execute("""PRAGMA synchronous=0""")
-        self.cur.execute("""PRAGMA locking_mode=EXCLUSIVE""")
-        # self.cur.execute("""PRAGMA journal_mode=TRUNCATE""")
-        self.cur.execute("""create table if not exists tiles (zoom_level integer, tile_column integer, tile_row integer, tile_data blob);""")
-        self.cur.execute("""create table if not exists metadata (name text, value text);""")
-        self.cur.execute("""create unique index if not exists name on metadata (name);""")
-        self.cur.execute("""create unique index if not exists tile_index on tiles (zoom_level, tile_column, tile_row);""")
-        metadata = [('name', setname), ('format', format2ext(self.format)),
-                    ('type', 'overlay' if overlay else 'baselayer'), ('version', version)]
+        self.cur.execute("PRAGMA synchronous=0")
+        self.cur.execute("PRAGMA locking_mode=EXCLUSIVE")
+      # self.cur.execute('PRAGMA journal_mode=TRUNCATE')
+        self.cur.execute("CREATE TABLE IF NOT EXISTS tiles (zoom_level integer, tile_column integer, tile_row integer, tile_data blob);")
+        self.cur.execute("CREATE TABLE IF NOT EXISTS metadata (name text, value text);")
+        self.cur.execute("CREATE UNIQUE INDEX IF NOT EXISTS name ON metadata (name);")
+        self.cur.execute("CREATE UNIQUE INDEX IF NOT EXISTS tile_index ON tiles (zoom_level, tile_column, tile_row);")
+
+        metadata = [("name", setname), ("format", format2ext(self.format)), ("type", "overlay" if overlay else "baselayer"), ("version", version)]
         if description:
-            metadata.append(('description', description))
-        for name, value in metadata:
-            self.cur.execute('insert or replace into metadata (name, value) values (?, ?)', (name, value))
+            metadata.append(("description", description))
+        self.cur.executemany("REPLACE INTO metadata (name, value) VALUES (?, ?)", metadata)
 
     def __str__(self):
         return "MBTilesWriter({0})".format(self.filename)
 
     def write_poly(self, poly):
         bbox = poly.bounds
-        self.cur.execute("""select value from metadata where name='bounds'""")
+        self.cur.execute("SELECT value FROM metadata WHERE name='bounds'")
         result = self.cur.fetchone
         if result:
-            b = result['value'].split(',')
+            b = result["value"].split(",")
             oldbbox = box(int(b[0]), int(b[1]), int(b[2]), int(b[3]))
             bbox = bbox.union(oldbbox).bounds
-        self.cur.execute("""insert or replace into metadata (name, value) values ('bounds', ?)""", ','.join(bbox))
+        self.cur.execute("REPLACE INTO metadata (name, value) VALUES ('bounds', ?)", ",".join(bbox))
```

Comment:
A single `REPLACE` is just a [compatibility feat](https://www.sqlite.org/lang_insert.html) for MySQL, and I suggest to revert it to the more readable `INSERT OR REPLACE`.